### PR TITLE
Allow finer grained chunking for Data Lake Store upload / download (#4014)

### PR DESCRIPTION
**Pull request by clehene**
Fixes https://github.com/Azure/azure-cli/issues/4014
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [x] The PR has modified HISTORY.rst with an appropriate description of the change (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### Command Guidelines

- [x] Each command and parameter has a meaningful description.
- [ ] Each new command has a test.

(see [Authoring Command Modules](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules))

### CODE REVIEW

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py [hunk 1]
```diff
@@ -175,6 +185,16 @@
         - name: --thread-count
           type: int
           short-summary: 'Optionally specify the parallelism of the download. Default is the number of cores in the local machine.'
+        - name: --chunk-size
+          type: int
+          short-summary: 'Number of bytes for a chunk. Large files are split into chunks. Files smaller than this number will always be transferred in a single thread. Default 268435456'
+        - name: --buffer-size
+          type: int
+          short-summary: 'Number of bytes for internal buffer. This block cannot be bigger than a chunk and cannot be smaller than a block. Default 4194304'
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_help.py]: No need to manually include `Default` in help like this. See my other comments.

File: src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py [hunk 2]
```diff
@@ -50,8 +50,14 @@
...
 register_cli_argument('dls fs download', 'overwrite', help='Indicates that, if the destination file or folder exists, it should be overwritten', action='store_true')
 register_cli_argument('dls fs download', 'thread_count', help='Specify the parallelism of the download. Default is the number of cores in the local machine.', type=int)
+register_cli_argument('dls fs download', 'chunk_size', help='Number of bytes for a chunk. Large files are split into chunks. Files smaller than this number will always be transferred in a single thread. Default 268435456', type=int)
+register_cli_argument('dls fs download', 'buffer_size', help='Number of bytes for internal buffer. This block cannot be bigger than a chunk and cannot be smaller than a block. Default 4194304 ', type=int)
+register_cli_argument('dls fs download', 'block_size', help='Number of bytes for a block. Within each chunk, we write a smaller block for each API call. This block cannot be bigger than a chunk. Default:4194304 ', type=int)
```
Comment (derekbekoe) [inline src/command_modules/azure-cli-dls/azure/cli/command_modules/dls/_params.py]: Can pass in `default` as a keyword argument and just specify the default value.
Then it wouldn't be needed in custom.py either.
This prevents duplicating the value in multiple places.

**Comment by begoldsm** [comment]
I am ok with this change, but I would caution that reducing the chunk-size can potentially have fragmentation implications when running jobs against that data in the future, since you will be concatenating many small files from potentially all over the distributed file system. It might be a good idea to put that as a small warning in the help.

**Comment by clehene** [comment]
@begoldsm that's a good point. I tried to get a better understanding of the relation between chunks and blocks, but couldn't find much. Also the ADLS WebHDFS API states `blocksize` as fixed to 256MB (https://docs.microsoft.com/en-us/rest/api/datalakestore/webhdfs-filesystem-apis)

Is there any good documentation to clear these?

**Comment by clehene** [comment]
@derekbekoe I implemented the changes, but it seems like github won't figure it out after rebasing...

**Comment by derekbekoe** [comment]
@clehene As CI is failing, run `check_style --ci` locally. There are some pylint/style issues currently.

**Comment by clehene** [comment]
@derekbekoe thanks, I just realized that additional changes meant for a different patch slipped. Checkstyle passes. Waiting for build to finish

**Comment by derekbekoe** [comment]
@clehene thanks for the PR!
 It will be in the nightly build tomorrow if you want to try it out.
https://github.com/Azure/azure-cli/#nightly-builds

**Pull request merged**